### ISSUE TITLE
Cannot unselect province in the choose a province screen or from the sheet menu

### ISSUE THREAD

**Issue by JuliannaR** [cds-snc/covid-shield-mobile#511]
Expected behaviour: To deselect by tapping the selected province. 

![Image from iOS (53)](https://user-images.githubusercontent.com/35537308/86849377-9450f500-c07d-11ea-8620-7fbf4e956bcd.png)

**Comment by timarney** [comment]
@jelen-HC  -  the new on-boarding flow will be merged shortly.
CC:
@smcmurtry

**Comment by timarney** [comment]
@JuliannaR 
@smcmurtry 

Per 
https://github.com/cds-snc/covid-shield-mobile/issues/249

These currently act like radio buttons 

Would a flat list more appropriate here https://reactnative.dev/docs/flatlist

**Comment by jelen-HC** [comment]
I have correct the issue for the current setup during the onboarding process. I am still sorting it out within the menu

**Comment by JuliannaR** [comment]
@timarney, @jelen-HC  I think a flatlist would enable selecting and unselecting of the same element without needing a skip or undefined,

**Comment by jelen-HC** [comment]
@timarney @JuliannaR The flatlist could work for selecting and unselecting of the same element, but you would need to essentially recreate a radio list to ensure only one element could be selected at a time or that a user wouldn't need to unselect an element to then select a different one. 

I have set a ternary statement for the onboarding Region select that basically looks to see if the region and item code match when a button is tapped, if true then it will set the region to none (allowing the user to unselect a region), otherwise it will select the new region the user taps on

**Comment by jelen-HC** [comment]
That is currently what I had sorted out and testing for the onboarding region select, with me currently still working on the menu region select. I will dig into this further and the flatlist
